dao/src/main/java/org/wso2/testgrid/dao/repository/TestPlanRepository.java

```diff
@@ -290,4 +295,120 @@ public TestPlan getLastFailure(TestPlan testPlan) {
                 .getResultList();
         return EntityManagerHelper.refreshResultList(entityManager, resultList);
     }
+
+    /**
+     * This method returns test failure summary for given test plan ids. (i.e for a given build job).
+     *
+     * @param testPlanIds test plan ids
+     * @return a List of {@link TestCaseFailureResultDTO} representing the test case failure results for a given build
+     */
+    public List<TestCaseFailureResultDTO> getTestFailureSummaryByTPId(List<String> testPlanIds)
+            throws TestGridDAOException {
+        StringBuilder sql = new StringBuilder("select failed_tc.test_name as name, failed_tc.failure_message as "
+                + "failureMessage, tp.infra_parameters as infraParametrs from test_plan tp join (select tc"
+                + ".test_name, tc.failure_message, ts.TESTPLAN_id  from test_case tc inner join test_scenario ts on "
+                + "ts.id=tc.TESTSCENARIO_id and tc.status = 'FAIL' and ts.TESTPLAN_id in (");
+        for (int i = 0; i < testPlanIds.size() - 1; i++) {
+            sql.append("?, ");
+        }
+        sql.append("?)) failed_tc on tp.id = failed_tc.TESTPLAN_id;");
+        Query query = entityManager.createNativeQuery(sql.toString());
+        int index = 1;
+        for (String s : testPlanIds) {
+            query.setParameter(index++, s);
+        }
+        return getResultList(query, TestCaseFailureResultDTO.class);
+    }
+
+    /**
+     * This method returns the test execution summary for given test plan ids(i.e for a given build job).
+     *
+     * @param testPlanIds test plan ids of a specific build job
+     * @return a List of {@link String} representing statuses of given test plans
+     */
+    public List<String> getTestExecutionSummaryByTPId(List<String> testPlanIds) {
```

Comment:
As I understand, this is a list of Statuses. Shall we use a method name like getTestPlanStatuses instead?

Reply:
Also, it would make much sense if we return a List<Status>.